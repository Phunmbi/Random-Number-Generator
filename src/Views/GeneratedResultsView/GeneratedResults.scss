.results {
  display: flex;
  height: 98vh;
  flex-direction: column;
  align-items: center;

  &-title{
    display: flex;
    justify-content: center;
    font-family: 'Hanalei', cursive;
    font-size: 40px;
    font-weight: 400;
    @media only screen and (max-width: 700px) {
      font-size: 18px;
    }
  }

  &-wrapper{
    display: flex;
    width: 80vw;
    flex-basis: auto;
    flex-wrap: wrap;
  }

  &-single-number{
    display: flex;
    justify-content: center;
    align-items: center;
    margin-right: 25px;
    width: 80px;
    height: 30px;
    border: 1px solid #c6b75d;
    font-size: 17px;
    font-weight: bold;
  }

  &-actions{
    position: fixed;
    bottom: 60px;
    display: flex;
    width: 260px;
    justify-content: space-between;
    align-items: center;
    div{
      cursor: pointer;
      display: flex;
      justify-content: space-between;
      padding: 5px;
      align-items: center;
      width: 140px;
      height: 30px;
      font-size: 19px;
      font-weight: bold;
      background-color: #c6b75d;
      border-radius: 5px;
      p {
        margin-top: 15px;
      }
    }
    div:first-child{
      width: 80px;
    }
  }
}
