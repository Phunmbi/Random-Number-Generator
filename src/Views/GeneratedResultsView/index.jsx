import React, {Component} from 'react';
import { generateNumbers } from '../../Helpers/GeneratorLogic';
import { saveToFile } from "../../Helpers/saveToFile";
import { sortAscending } from "../../Helpers/sortAscending";
import { sortDescending } from "../../Helpers/sortDescending";
import Home from './Images/Home.svg';
import Upload from './Images/Upload.svg';
import Ascending from './Images/Ascending.svg';
import Descending from './Images/Descending.svg';
import './GeneratedResults.scss';

class GeneratedResults extends Component {
  state = {
    numbers: [],
    maxNumber: null,
    minNumber: null
  };

  componentDidMount() {
    const generatedNumbersAutoSortedAscending = sortAscending(
      generateNumbers(Number(this.props.location.search.slice(1, this.props.location.search.length)))
    );

    this.setState({
      numbers: generatedNumbersAutoSortedAscending,
      minNumber: generatedNumbersAutoSortedAscending[0],
      maxNumber: generatedNumbersAutoSortedAscending[generatedNumbersAutoSortedAscending.length - 1],
    })
  }

  SingleGeneratedNumber = (int, index) => {
    return (
      <div key={index}>
        {index + 1}
        <p className="results-single-number">{int}</p>
      </div>
    )
  };

  handleAscendingSort = () => {
    this.setState({
      numbers: sortAscending(this.state.numbers)
    })
  };

  handleDescendingSort = () => {
    this.setState({
      numbers: sortDescending(this.state.numbers)
    })
  };

  renderSorting = () => {
    return (
      <div className="results-sort">
        <div onClick={() => this.handleAscendingSort()} className="results-sort__ascending">
          <img src={Ascending} alt="Ascending" />
          <p>Sort Ascending</p>
        </div>
        <div onClick={() => this.handleDescendingSort()} className="results-sort__descending">
          <img src={Descending} alt="Descending" />
          <p>Sort Descending</p>
        </div>
        <div className="results-sort__Total">
          <h3>Total numbers:</h3>
          <p>{this.state.numbers.length}</p>
        </div>
        <div className="results-sort__Min">
          <h3>Minimum Number:</h3>
          <p>{this.state.minNumber}</p>
        </div>
        <div className="results-sort__Maxa">
          <h3>Maximum Number:</h3>
          <p>{this.state.maxNumber}</p>
        </div>
      </div>
    )
  };

  render() {
    return (
      <div className="results">
        <div className="results-title">
          <h2>Generated Numbers</h2>
        </div>
        {this.renderSorting()}
        <div className="results-wrapper">
          {this.state.numbers.map((each, index) => {
            return this.SingleGeneratedNumber(each, index);})}
        </div>
        <div className="results-actions">
          <div className="results-actions__redirect" onClick={() => this.props.history.push('/')} >
            <img src={Home} alt="Home"/>
            <p>Home</p>
          </div>
          <div className="results-actions__upload" onClick={() => saveToFile(this.state.numbers)}>
            <img src={Upload} alt="Save Numbers" />
            <p>Save Numbers</p>
          </div>
        </div>
      </div>
    );
  }
}

export default GeneratedResults;
