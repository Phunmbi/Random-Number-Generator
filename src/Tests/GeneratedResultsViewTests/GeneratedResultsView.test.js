import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import GeneratedResultsView from '../../Views/GeneratedResultsView';

const props = {
  location: {
    search: '?10',
  },
  history: {
    push: jest.fn(),
  }
};

let wrapped;

beforeEach(() => {
  wrapped = mount (
    <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
      <GeneratedResultsView {...props} />
    </MemoryRouter>
  );
});

it('should take screenshot successfully', () => {
  expect(wrapped).toMatchSnapshot();
});

it('should simulate sort by ascending', () => {
  wrapped.find('.results-sort__ascending').simulate('click');
  expect(wrapped.find('GeneratedResultsView').exists()).toEqual(false);
});

it('should simulate sort by descending', () => {
  wrapped.find('.results-sort__descending').simulate('click');
  expect(wrapped.find('GeneratedResultsView').exists()).toEqual(false);
});

it('should simulate home redirect', () => {
  wrapped.find('.results-actions').find('.results-actions__redirect').simulate('click');
  expect(wrapped.find('GeneratedResultsView').exists()).toEqual(false);
});
