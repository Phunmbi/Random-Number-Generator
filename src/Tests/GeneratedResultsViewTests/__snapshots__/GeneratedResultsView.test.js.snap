// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take screenshot successfully 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testkey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testkey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testkey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <GeneratedResults
      history={
        Object {
          "push": [MockFunction],
        }
      }
      location={
        Object {
          "search": "?10",
        }
      }
    >
      <div
        className="results"
      >
        <div
          className="results-title"
        >
          <h2>
            Generated Numbers
          </h2>
        </div>
        <div
          className="results-sort"
        >
          <div
            className="results-sort__ascending"
            onClick={[Function]}
          >
            <img
              alt="Ascending"
              src="Ascending.svg"
            />
            <p>
              Sort Ascending
            </p>
          </div>
          <div
            className="results-sort__descending"
            onClick={[Function]}
          >
            <img
              alt="Descending"
              src="Descending.svg"
            />
            <p>
              Sort Descending
            </p>
          </div>
          <div
            className="results-sort__Total"
          >
            <h3>
              Total numbers:
            </h3>
            <p>
              10
            </p>
          </div>
          <div
            className="results-sort__Min"
          >
            <h3>
              Minimum Number:
            </h3>
            <p>
              0128225234
            </p>
          </div>
          <div
            className="results-sort__Maxa"
          >
            <h3>
              Maximum Number:
            </h3>
            <p>
              0915417426
            </p>
          </div>
        </div>
        <div
          className="results-wrapper"
        >
          <div
            key="0"
          >
            1
            <p
              className="results-single-number"
            >
              0128225234
            </p>
          </div>
          <div
            key="1"
          >
            2
            <p
              className="results-single-number"
            >
              0197359795
            </p>
          </div>
          <div
            key="2"
          >
            3
            <p
              className="results-single-number"
            >
              0224627479
            </p>
          </div>
          <div
            key="3"
          >
            4
            <p
              className="results-single-number"
            >
              0336851176
            </p>
          </div>
          <div
            key="4"
          >
            5
            <p
              className="results-single-number"
            >
              0520682485
            </p>
          </div>
          <div
            key="5"
          >
            6
            <p
              className="results-single-number"
            >
              0691708939
            </p>
          </div>
          <div
            key="6"
          >
            7
            <p
              className="results-single-number"
            >
              0702670157
            </p>
          </div>
          <div
            key="7"
          >
            8
            <p
              className="results-single-number"
            >
              0809795710
            </p>
          </div>
          <div
            key="8"
          >
            9
            <p
              className="results-single-number"
            >
              0885478393
            </p>
          </div>
          <div
            key="9"
          >
            10
            <p
              className="results-single-number"
            >
              0915417426
            </p>
          </div>
        </div>
        <div
          className="results-actions"
        >
          <div
            className="results-actions__redirect"
            onClick={[Function]}
          >
            <img
              alt="Home"
              src="Home.svg"
            />
            <p>
              Home
            </p>
          </div>
          <div
            className="results-actions__upload"
            onClick={[Function]}
          >
            <img
              alt="Save Numbers"
              src="Upload.svg"
            />
            <p>
              Save Numbers
            </p>
          </div>
        </div>
      </div>
    </GeneratedResults>
  </Router>
</MemoryRouter>
`;
